src/rez/version/_requirement.py

```diff
@@ -264,15 +270,17 @@ def conflicts_with(self, other):
                 return other.range_.issuperset(self.range_)
             else:
                 return not self.range_.intersects(other.range_)
-        else:  # VersionedObject
+        elif isinstance(other, VersionedObject):
             if (self.name_ != other.name_) or (self.range is None):
                 return False
             if self.conflict:
                 return (other.version_ in self.range_)
             else:
                 return (other.version_ not in self.range_)
+        else:
+            return NotImplemented
```

Comment:
Should this raise an `NotImplementedError` instead?

Reply:
Nope.   `NotImplemented` is a singleton that is used to indicate "I don't support this comparison", and it will fall back to other object to see if it supports it. 

https://docs.python.org/3/reference/datamodel.html#object.__ge__

> When no appropriate method returns any value other than [NotImplemented](https://docs.python.org/3/library/constants.html#NotImplemented), the == and != operators will fall back to is and is not, respectively.

This is a fundamental part of how comparison works in python, and returning `NotImplemented` is the most correct behavior.

Reply:
But should we allow comparing with anything else than a `Requirement` or `VersionedObject`? I wouldn't consider this function a comparator per say.